lib/ogr/feature.rb

```diff
@@ -1,11 +1,12 @@
 # frozen_string_literal: true
 
-require 'date'
 require_relative '../ogr'
 require_relative '../gdal'
 
 module OGR
   class Feature
+    autoload :DateTimeConv, File.expand_path('feature/date_time_conv', __dir__ || '')
```

Comment:
the `|| ''` here is for type checking. There's some unexpected things with `expand_path` and `__dir__`. `__dir__` returns either a `String` or `nil`, and the 2nd param for `expand_path` has to be either a `String` or an object that responds to `#to_path`.

```
lib/ogr/feature.rb:8:71: [error] Cannot pass a value of type `(::String | nil)` as an argument of type `(::string | ::_ToPath)`
│   (::String | nil) <: (::string | ::_ToPath)
│     nil <: (::string | ::_ToPath)
│       nil <: ::string
│         nil <: (::String | ::_ToStr)
│           nil <: ::String
│
│ Diagnostic ID: Ruby::ArgumentTypeMismatch
│
└     autoload :DateTimeConv, File.expand_path('feature/date_time_conv', __dir__)
                                                                         ~~~~~~~
```